packages/models/src/errors.ts

```diff
@@ -86,6 +86,16 @@ export const ErrorTypes = {
     httpStatus: 404,
     title: "Agreement not found",
   },
+  OperationNotAllowed: {
+    code: "0007",
+    httpStatus: 400,
+    title: "Operation not allowed",
+  },
+  AgreementNotInExpectedState: {
```

Comment:
any reason why errors of a specific service are under the shared package?
it seems very difficult to know which error is used where and if there are any duplicates.

It's ok for me not changing it in this PR, but we should discuss this

Reply:
For sure! We should refactor how we handle the errors